Library/Homebrew/cmd/uninstall.rb

```diff
@@ -66,11 +68,13 @@ def uninstall
     )
 
     if args.zap?
-      T.unsafe(Cask::Cmd::Zap).zap_casks(
-        *casks,
-        verbose: args.verbose?,
-        force:   args.force?,
-      )
+      casks.each do |cask|
+        odebug "Zapping Cask #{cask}"
+
+        raise Cask::CaskNotInstalledError, cask if !cask.installed? && !args.force?
+
+        Cask::Installer.new(cask, verbose: args.verbose?, force: args.force?).zap
+      end
```

Comment:
I wonder if it'd be better to just have this in another file like `Cask::Uninstall`. Then, we could provide a similar interface to what we do with uninstalling kegs above and combine the zap and uninstall command logic a bit.

```rb
Cask::Uninstall.uninstall_casks(
  *casks,
  verbose: args.verbose?,
  force:   args.force?,
  zap: args.zap?,
)
```

Reply:
@apainintheneck I dunno. I kinda like this with how minimal it is and avoiding the need for another file.

Reply:
In this case, it's just a few lines but once `cask/cmd/uninstall` gets removed a few more lines will be added. It's not the end of the word either way but I'd prefer to keep cask stuff in `/cask` and keep the commands leaner if possible. Another option would be to add them to the `Uninstall` module but that's not my first choice either.

Reply:
Let's just do this as-is now and require that change in `cask/cmd/uninstall`.